homeassistant/components/vicare/climate.py

```diff
@@ -0,0 +1,363 @@
+        return self._support_flags
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current hvac mode."""
+        return VICARE_TO_HA_HVAC_HEATING.get(self._current_mode, STATE_UNKNOWN)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set a new hvac mode on the ViCare API."""
+        _LOGGER.info(
+            "setting hvac mode to %s (%s)",
+            hvac_mode,
+            HA_TO_VICARE_HVAC_HEATING.get(hvac_mode, STATE_UNKNOWN),
+        )
+        self._api.setMode(HA_TO_VICARE_HVAC_HEATING.get(hvac_mode, STATE_UNKNOWN))
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac modes."""
+        return list(HA_TO_VICARE_HVAC_HEATING.keys())
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(
+            VICARE_TEMP_HEATING_MIN, TEMP_CELSIUS, self.temperature_unit
+        )
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return convert_temperature(
+            VICARE_TEMP_HEATING_MAX, TEMP_CELSIUS, self.temperature_unit
+        )
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        self._api.setProgramTemperature(self._current_program, self._target_temperature)
+        self.schedule_update_ha_state()
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self._current_program == VICARE_PROGRAM_COMFORT:
+            return PRESET_COMFORT
+        if self._current_program == VICARE_PROGRAM_ECO:
+            return PRESET_ECO
+        return None
+
+    @property
+    def preset_modes(self):
+        """Return the available preset mode."""
+        return [PRESET_COMFORT, PRESET_ECO]
+
+    def set_preset_mode(self, preset_mode):
+        """Set new preset mode and deactivate any existing programs."""
+        self._api.deactivateProgram(self._current_program)
+
+        if preset_mode == PRESET_COMFORT:
+            self._api.activateProgram(VICARE_PROGRAM_COMFORT)
```

Comment:
Make two constant dicts between the vicare programs and home assistant presets and use the dicts here and above in the preset mode property.